src/Layers/xrRender_R2/r2_loader.cpp

```diff
@@ -311,32 +304,55 @@ void CRender::LoadSectors(IReader* fs)
 {
     // allocate memory for portals
     const u32 size = fs->find_chunk(fsL_PORTALS);
-    R_ASSERT(0 == size % sizeof(CPortal::b_portal));
-
-    const u32 count = size / sizeof(CPortal::b_portal);
-    Portals.resize(count);
+    R_ASSERT(0 == size % sizeof(CPortal::level_portal_data_t));
 
-    for (u32 c = 0; c < count; c++)
-        Portals[c] = xr_new<CPortal>();
+    const u32 portals_count = size / sizeof(CPortal::level_portal_data_t);
+    xr_vector<CPortal::level_portal_data_t> portals_data{portals_count};
 
     // load sectors
+    xr_vector<CSector::level_sector_data_t> sectors_data;
+    
+    float largest_sector_vol = 0.0f;
     IReader* S = fs->open_chunk(fsL_SECTORS);
     for (u32 i = 0;; i++)
     {
         IReader* P = S->open_chunk(i);
         if (!P)
             break;
+        
+        auto& sector_data = sectors_data.emplace_back();
+        {
+            u32 size = P->find_chunk(fsP_Portals);
+            R_ASSERT(0 == (size & 1));
+            u32 portals_in_sector = size / sizeof(u16);
+
+            sector_data.portals_id.reserve(portals_in_sector);
+            while (portals_in_sector)
+            {
+                const u16 ID = P->r_u16();
+                sector_data.portals_id.emplace_back(ID);
+                --portals_in_sector;
+            }
 
-        CSector* __S = xr_new<CSector>();
-        __S->load(*P);
-        Sectors.push_back(__S);
+            size = P->find_chunk(fsP_Root);
+            R_ASSERT(size == 4);
+            sector_data.root_id = P->r_u32();
 
+            // Find the larget sector
```

Comment:
Original GSC comment
```suggestion
            // Search for default sector - assume "default" or "outdoor" sector is the largest one
            //. hack: need to know real outdoor sector
```